useful-scripts/misc-test-utils/src/services/publicationService.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-console */
+import axios from 'axios';
+import chalk from 'chalk';
+import { v4 } from 'uuid';
+import errorHandler from '../utils/errorHandler';
+import { ReleaseData } from '../types/ReleaseData';
+import sleep from '../utils/sleep';
+
+const { ADMIN_URL, TOPIC_ID, JWT_TOKEN } = process.env;
+
+const publicationService = {
+  // eslint-disable-next-line consistent-return
+  createPublication: async () => {
+    console.time('createPublication');
+
+    try {
+      const res = await axios({
+        method: 'POST',
+        url: `${ADMIN_URL}/api/publications`,
+        data: {
+          title: `importer-testing-${v4()}`,
+          topicId: TOPIC_ID,
+          contact: {
+            teamName: 'testing',
+            teamEmail: 'johndoe@gmail.com',
+            contactName: 'John Doe',
+            contactTelNo: '123456789',
+          },
+        },
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${JWT_TOKEN}`,
+        },
+      });
```

Comment:
We can refactor all of the Axios invocations with an actual Axios instance that's pre-configured with the correct headers (so we don't have to repeat ourselves):

```ts
const adminApi = axios.create({
  baseURL: ADMIN_URL,
  headers: {
    'Content-Type': 'application/json',
    Authorization: `Bearer ${JWT_TOKEN}`,
  },
});

const publicationService = {
  createPublication: async () => {
    const res = adminApi.post(`/api/publications`, { .... });
    // etc
  },
};
```

This is basically what we do with all of our frontend services :wink:

Reply:
cool. Initially, I took a look at the content API service but I didn't go forward with implementing an admin API instance as I wasn't sure what the param serializer was doing: 
```
export const contentApi = new Client(
  axios.create({
    baseURL: process.env.CONTENT_API_BASE_URL,
    paramsSerializer: commaSeparated,
  }),
);
```

Reply:
The param serializer is for query parameters. In the early days of the project, we tried to put the table tool query options inside of the URL query parameters. The table builder endpoint also used to be a GET.

This didn't work too well because we ended up with urls like `location[]=london&location[]=barnsley&...`, so we added the comma separated serializer so that we get query strings like `location=london,barnsley` instead (way more compact).

We didn't end up sticking with the GET endpoint though as it would lead to potentially huge query strings which could easily hit the URL character limit in browsers. We eventually turned it into a POST endpoint so that we could submit everything in the request body instead.

The comma separated serializer just stuck around after that really :smile:

I'd advise that you also use it in any of your own Axios instances too!